searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp

```diff
@@ -253,6 +255,63 @@ Test::testThatOptimizePreservesUnpack()
     fixup_bitvectors();
 }
 
+void verifyOrUnpack(SearchIterator & s, const TermFieldMatchData * tfmd) {
+    s.initFullRange();
+    s.seek(1);
+    for (size_t i = 0; i < 3; i++) {
+        EXPECT_EQUAL(0u, tfmd[0].getDocId());
+    }
+    s.unpack(1);
+    EXPECT_EQUAL(0u, tfmd[0].getDocId());
+    EXPECT_EQUAL(1u, tfmd[1].getDocId());
+    EXPECT_EQUAL(0u, tfmd[2].getDocId());
+}
+
+void
+Test::testUnpackOfOr() {
+    _bvs[0]->clearBit(1);
+    _bvs[1]->setBit(1);
+    _bvs[2]->clearBit(1);
+    UnpackInfo all;
+    all.forceAll();
+    verifyUnpackOfOr(all);
+
+    UnpackInfo unpackInfo;
+    unpackInfo.add(1);
+    unpackInfo.add(2);
+    verifyUnpackOfOr(unpackInfo);
+
+    fixup_bitvectors();
+}
+
+void
+Test::verifyUnpackOfOr(const UnpackInfo &unpackInfo)
+{
+    TermFieldMatchData tfmdA[3];
+    MultiSearch::Children children;
+    children.push_back(createIter(0, false, tfmdA[0], false).release());
+    children.push_back(createIter(1, false, tfmdA[1], false).release());
+    children.push_back(createIter(2, false, tfmdA[2], false).release());
+    SearchIterator::UP s(OrSearch::create(children, false, unpackInfo));
+    verifyOrUnpack(*s, tfmdA);
+
+    for (auto & tfmd : tfmdA) {
+        tfmd.resetOnlyDocId(0);
+    }
+
+    const MultiSearch * ms = dynamic_cast<const MultiSearch *>(s.get());
+    EXPECT_TRUE(ms != nullptr);
+    EXPECT_EQUAL(3u, ms->getChildren().size());
+
+    s = MultiBitVectorIteratorBase::optimize(std::move(s));
+    s->initFullRange();
+    ms = dynamic_cast<const MultiSearch *>(s.get());
```

Comment:
consider checking for multi-bitvector optimization

Reply:
Fixed